hadoop-ozone/ozone-manager/src/main/java/org/apache/hadoop/ozone/om/ListIterator.java

```diff
@@ -321,7 +332,9 @@ public boolean hasNext() {
       return !minHeap.isEmpty();
     }
 
-    public HeapEntry next() {
+    // HeapEntry can be null when a key is marked as deleted because
```

Comment:
I didn't quite get this comment. Does it mean for deleted entries cache will have null value? If that's the case, I think better approach would be to filter it out when populating CacheIter in `populateCacheMap`.

Reply:
This information is required while checking db entry, as key is deleted in cache but not yet flushed to db, so db can have these entry. To avoid returning db entry, can not filter out.

Reply:
Yeah we need the null values